## Research Question

"How do chronic disease prevalence and health disparities vary by socioeconomic factors (e.g., income inequality, unemployment) across U.S. states?"

## Problem Statement 

Chronic diseases disproportionately affect different socioeconomic groups across the U.S. states. Factors such as income inequality, employment status, and healthcare access contribute to these disparities. This study aims to analyze trends in chronic disease prevalence and identify correlations with socioeconomic indicators. We hypothesize that states with higher income inequality and unemployment will exhibit higher chronic disease rates and lower access to healthcare

## Data Description 

https://catalog.data.gov/dataset/u-s-chronic-disease-indicators   The U.S. Chronic Disease Indicators (CDI) dataset is provided by the CDC’s Division of Population Health. It consists of 309,215 records covering 115 key health indicators related to chronic diseases across all U.S. states and territories. This dataset is designed to provide a standardized approach for collecting, analyzing, and reporting chronic disease data to support public health decision-making.

## Import the libraries

In [12]:
# Import the libraries
import numpy as np                  # Scientific Computing
import pandas as pd                 # Data Analysis
import matplotlib.pyplot as plt     # Plotting
import seaborn as sns               # Statistical Data Visualization

# Let's make sure pandas returns all the rows and columns for the dataframe
pd.set_option('display.max_rows', None)
pd.set_option('display.max_columns', None)

# Force pandas to display full numbers instead of scientific notation
# pd.options.display.float_format = '{:.0f}'.format

# Library to suppress warnings
import warnings
warnings.filterwarnings('ignore')

### Read the dataset

In [13]:
# Read the dataset
path = pd.read_csv('U.S._Chronic_Disease_Indicators.csv')

# Create the Dataframe
us_chronic_desease = pd.DataFrame(path)


### Understanding the Dataset

- Checking first elements of the DataFrame with .head( ) method
- After you run the code above, nothing will appear.
- So you have to write df to see your data. But instead of seeing all the data, we are going to use the “.head( )” method to see the first five elements of the data.
- Before you run the read_csv code, you can write df.head( ) below. So it’s going to be like this:

Inside the parentheses, we can write the number of elements that we want to see.
If we leave it blank, it will show the first five elements.
If we write 7 inside of the parentheses, it will show the first 7 elements of the dataframe.

In [15]:
# Display the first ten rows of the Dataframe
us_chronic_desease.head(7)

YearStart  YearEnd LocationAbbr LocationDesc DataSource          Topic  \
0       2019     2019           AR     Arkansas      BRFSS       Diabetes   
1       2019     2019           ID        Idaho      BRFSS       Diabetes   
2       2019     2019           IN      Indiana      YRBSS          Sleep   
3       2019     2019           IA         Iowa       NVSS         Asthma   
4       2019     2019           IA         Iowa      BRFSS         Asthma   
5       2019     2019           IA         Iowa       NVSS       Diabetes   
6       2019     2019           IA         Iowa      BRFSS  Health Status   

                                            Question  Response DataValueUnit  \
0                              Diabetes among adults       NaN             %   
1                              Diabetes among adults       NaN             %   
2    Short sleep duration among high school students       NaN             %   
3  Asthma mortality among all people, underlying ...       NaN        Number   
4                        Current asthma among adults       NaN             %   
5  Diabetes mortality among all people, underlyin...       NaN        Number   
6            Recent activity limitation among adults       NaN        Number   

      DataValueType  DataValue  DataValueAlt DataValueFootnoteSymbol  \
0  Crude Prevalence       13.6          13.6                     NaN   
1  Crude Prevalence       10.6          10.6                     NaN   
2  Crude Prevalence        NaN           NaN                       *   
3            Number       54.0          54.0                     NaN   
4  Crude Prevalence       10.3          10.3                     NaN   
5            Number       54.0          54.0                     NaN   
6        Crude Mean        2.3           2.3                     NaN   

   DataValueFootnote  LowConfidenceLimit  HighConfidenceLimit  \
0                NaN                12.1                 15.4   
1                NaN                 9.1                 12.2   
2  No data available                 NaN                  NaN   
3                NaN                 NaN                  NaN   
4                NaN                 9.1                 11.7   
5                NaN                 NaN                  NaN   
6                NaN                 2.1                  2.5   

  StratificationCategory1 Stratification1  StratificationCategory2  \
0                     Sex            Male                      NaN   
1                     Sex            Male                      NaN   
2                   Grade        Grade 12                      NaN   
3                 Overall         Overall                      NaN   
4                     Age       Age 18-44                      NaN   
5                     Age        Age 0-44                      NaN   
6                     Sex          Female                      NaN   

   Stratification2  StratificationCategory3  Stratification3  \
0              NaN                      NaN              NaN   
1              NaN                      NaN              NaN   
2              NaN                      NaN              NaN   
3              NaN                      NaN              NaN   
4              NaN                      NaN              NaN   
5              NaN                      NaN              NaN   
6              NaN                      NaN              NaN   

                                     Geolocation  LocationID TopicID  \
0   POINT (-92.27449074299966 34.74865012400045)           5     DIA   
1  POINT (-114.3637300419997 43.682630005000476)          16     DIA   
2  POINT (-86.14996019399968 39.766910452000445)          18    SLEP   
3   POINT (-93.81649055599968 42.46940091300047)          19     AST   
4   POINT (-93.81649055599968 42.46940091300047)          19     AST   
5   POINT (-93.81649055599968 42.46940091300047)          19     DIA   
6   POINT (-93.81649055599968 42.46940091300047)          19     HEA 

Checking last elements of the DataFrame with .tail() method
There is also a method to see the see last n number of elements.
The method is called .tail().
The same rule is also applied here. If we leave the parentheses blank, it will be set as 5, if we write 7 inside of the parentheses, it will show the last 7 elements of the dataframe.

In [17]:
# Display the last ten rows of the dataframe
# Syntax: DataFrame.tail(qty)
us_chronic_desease.tail(7)

YearStart  YearEnd LocationAbbr    LocationDesc DataSource  \
309208       2022     2022           VT         Vermont      BRFSS   
309209       2022     2022           VI  Virgin Islands      BRFSS   
309210       2022     2022           VI  Virgin Islands      BRFSS   
309211       2022     2022           WV   West Virginia      BRFSS   
309212       2022     2022           WI       Wisconsin      BRFSS   
309213       2022     2022           VT         Vermont      BRFSS   
309214       2022     2022           WA      Washington      BRFSS   

                                        Topic  \
309208                                  Sleep   
309209                           Immunization   
309210                                Tobacco   
309211  Chronic Obstructive Pulmonary Disease   
309212                           Immunization   
309213          Social Determinants of Health   
309214                                Alcohol   

                                                 Question  Response  \
309208                  Short sleep duration among adults       NaN   
309209                 Influenza vaccination among adults       NaN   
309210  Quit attempts in the past year among adult cur...       NaN   
309211  Chronic obstructive pulmonary disease among ad...       NaN   
309212  Pneumococcal vaccination among adults aged 65 ...       NaN   
309213  Lack of health insurance among adults aged 18-...       NaN   
309214             Binge drinking prevalence among adults       NaN   

       DataValueUnit            DataValueType  DataValue  DataValueAlt  \
309208             %         Crude Prevalence       26.5          26.5   
309209             %         Crude Prevalence       34.2          34.2   
309210             %  Age-adjusted Prevalence        NaN           NaN   
309211             %         Crude Prevalence       14.0          14.0   
309212             %         Crude Prevalence       64.2          64.2   
309213             %         Crude Prevalence        NaN           NaN   
309214             %  Age-adjusted Prevalence       19.2          19.2   

       DataValueFootnoteSymbol  \
309208                     NaN   
309209                     NaN   
309210                       #   
309211                     NaN   
309212                     NaN   
309213                    ****   
309214                     NaN   

                                        DataValueFootnote  LowConfidenceLimit  \
309208                                                NaN                16.8   
309209                                                NaN                24.2   
309210  No data available for this indicator because t...                 NaN   
309211                                                NaN                12.8   
309212                                                NaN                52.2   
309213  Data suppressed; denominator < 50 or relative ...                 NaN   
309214                                                NaN                18.3   

        HighConfidenceLimit StratificationCategory1  \
309208                 39.3          Race/Ethnicity   
309209                 45.8                     Age   
309210                  NaN          Race/Ethnicity   
309211                 15.2                 Overall   
309212                 74.6          Race/Ethnicity   
309213                  NaN          Race/Ethnicity   
309214                 20.2                     Sex   

                                       Stratification1  \
309208                             Asian, non-Hispanic   
309209                                        Age >=65   
309210  American Indian or Alaska Native, non-Hispanic   
309211                                         Overall   
309212                             Black, non-Hispanic   
309213      Hawaiian or Pacific Islander, non-Hispanic   
309214                                            Male   

        StratificationCategory2  Stratification2  Stratification

In [18]:
# display the dimensions of the data
# This is the number of rows and columns in the data
# Syntax: DataFrame.shape
us_chronic_desease.shape

(309215, 34)

In [19]:
309215*34

10513310

* State the shape of the dataframe :
  - The dataframe have 309215 rows
  - The dataframe have 34 Columns
  - There are 10513310 total datapoints observed i my dataset.

### Data Types
* The main data types in Pandas dataframes are `object`, `float`, `int64`, `bool` and `datetime64`. In order to understand each attribute of the data, it is always good to know the data type of each column.

##### `.info()` method
* This method prints information about a DataFrame including the index `dtype` and column dtypes, non-null values and memory usage.

In [20]:
# Let's check the basic information about the dataset
# Syntax: DataFrame.info()
us_chronic_desease.info(show_counts=True, verbose=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 309215 entries, 0 to 309214
Data columns (total 34 columns):
 #   Column                     Non-Null Count   Dtype  
---  ------                     --------------   -----  
 0   YearStart                  309215 non-null  int64  
 1   YearEnd                    309215 non-null  int64  
 2   LocationAbbr               309215 non-null  object 
 3   LocationDesc               309215 non-null  object 
 4   DataSource                 309215 non-null  object 
 5   Topic                      309215 non-null  object 
 6   Question                   309215 non-null  object 
 7   Response                   0 non-null       float64
 8   DataValueUnit              309215 non-null  object 
 9   DataValueType              309215 non-null  object 
 10  DataValue                  209196 non-null  float64
 11  DataValueAlt               209196 non-null  float64
 12  DataValueFootnoteSymbol    101716 non-null  object 
 13  DataValueFootnote          10

### Observations of the Data Set
Describe the dataset.
- There are 34 rows and 309215 columns.
- The data types are integer , object ,float 
Based on the number of expected data points, and those listed by the `.info()` method, how many missing (null) values are there?
- There are 3953604 missing (null) values 

The number of non-null values does not match the total number expected based on the number of rows in the dataframe. This indicates the presence of missing values that will need further investigation.



# Data Cleaning:


## Data Analysis and Preparation
The steps of analysis and preparation of the data for statistical modeling has several steps:
1. Check dimensions of the dataframe to determine the number rows and columns. This increases understanding of the data structure and size.
2. Check data types, ensuring the data types are correct. Refer data definitions to validate the results. For example dates including year, month, and day should be updated from integers or strings to `datetime` for ease of use in time series analysis.
3. Update data types based on the business definition, changing the data types as per requirement.
4. Using Python and Numpy methods, examine the summary statistics. This assists in determination of the data scale, including the relative minima and maxima for the range. The distribution of data with early determination of skewness or kurtosis are also visible for outliers in the data.
5. Checking for missing values that may cause noise in machine learning models and abberations in visualizations.
6. Study the correlation between data variables for key insights and future feature engineering.
7. Detection of outliers that may contribute to the skewness in data or add to the kurtosis affecting the data ranges.

### Headers Update and Map New Column

>
- To aid in analysis and visualization add a new columns as needed. For instance instead of an entire state name, add a column that has a two letter abbreviation for the state.
- The list of headers requires standardization, these will be updated ensure uniformity.
- Update ALL CAPS or all lowercase to the appropriate case. These can be updated to Title Case.

In [30]:
# Let's create a list of the columns in the dataset
# Use the variable = DataFrame.columns method
chroniccols=us_chronic_desease.columns
chroniccols

Index(['YearStart', 'YearEnd', 'LocationAbbr', 'LocationDesc', 'DataSource',
       'Topic', 'Question', 'Response', 'DataValueUnit', 'DataValueType',
       'DataValue', 'DataValueAlt', 'DataValueFootnoteSymbol',
       'DataValueFootnote', 'LowConfidenceLimit', 'HighConfidenceLimit',
       'StratificationCategory1', 'Stratification1', 'StratificationCategory2',
       'Stratification2', 'StratificationCategory3', 'Stratification3',
       'Geolocation', 'LocationID', 'TopicID', 'QuestionID', 'ResponseID',
       'DataValueTypeID', 'StratificationCategoryID1', 'StratificationID1',
       'StratificationCategoryID2', 'StratificationID2',
       'StratificationCategoryID3', 'StratificationID3'],
      dtype='object')

In [ ]:
# Let's Update the Headers for Syntax Consistency
# Syntax: df = df.rename(columns={'currentColumnName':'newColumnName', 'nextCurrentColumnName':'nextNewColumnName'})


# Let's view the new columns and update the variable
# Pass the columns to the variable: Use the variable = DataFrame.columns method

# Call the variable to see the contents